src/rust/ide/view/graph-editor/src/selection.rs

```diff
@@ -0,0 +1,510 @@
+//! Module that contains the logic sor selecting nodes. This includes selecting single nodes
+//! by clicking on them separately, as well as click+drag for selecting with a selection area.
+use ensogl::prelude::*;
+
+use ensogl::frp;
+use ensogl::gui::cursor::Cursor;
+use ensogl::gui::cursor;
+
+use crate::NodeId;
+use crate::Nodes;
+use crate::TouchState;
+
+
+
+// ============
+// === Mode ===
+// ============
+
+/// Possible selection modes.
+#[derive(Clone,Copy,Debug,Eq,PartialEq)]
+pub enum Mode {
+    /// Select a single node when clicking on it. Deselects all other nodes.
+    Normal,
+    /// Toggle the selection state of the select node without changing the selection state of
+    ///other node. This allows to add and remove nodes from the current selection.
+    Multi,
+    /// Add selected nodes to the set of currently selected nodes.
+    Merge,
+    /// Remove selected nodes from the set of currently selected nodes.
+    Subtract,
+    /// Invert the selection state of the selected nodes.
+    Inverse
+}
+
+impl Mode {
+    fn single_should_select(self, was_selected:bool) -> bool {
```

Comment:
I don't understand what these functions do. What `single`/`multi` means in the name, and how `multi` in names is related to `Multi` mode?